### PR TITLE
AC-1937 - Fix form submit events in the places it is not triggering

### PR DESCRIPTION
**Pull request by shrirupa**
AC-1937 - Fix form submit events in the places it is not triggering

### What Changed

- In BackupCodeModal, moved the onClick from Button to Form's handleSubmit and updated the unit test accordingly.
- In EnableTfaForm and DisableTfa form made the same change as above.
-----------------------------------------------------------------------------
- In InsertSnippetModal, removed the onClick from submit button and updated the test accordingly

### How To Test
Before testing, make a small code change, navigate src/components/tracking/Form.js and change debug = true. This will enable you to see console logs when you submit forms.
- For testing EnableTfaModal, navigate to /account/profile and click Enable Tfa and submit the opened form. Check if you see logs for fields and form submitted.
- For testing BackupCodeModal and DisableTfaModal, create an account and Enable Tfa on the account form /account/profile page.  After enabling Tfa you should be able to see Generate Backup Codes and Disable Tfa buttons in the Two Factor Authentication section. Open both Modals and submit the respective forms and check logs for fields and form submitted.

--------------------------------------------------------------------------------
- For testing InsertSnippetModal, navigate to /templates/edit/bob-test/draft/content, click on horizontal ellipsis icon and Click on Insert Snippet menu item.

### To Do

- [ ] Address feedback

### CODE REVIEW

File: src/pages/profile/components/BackupCodesModal.js [hunk 1]
```diff
@@ -34,57 +34,39 @@ export default class BackupCodesModal extends Component {
...
-                  <span role="img" aria-label="phone in toilet emojis">
-                    📱&nbsp;➡︎&nbsp;🚽
-                  </span>{' '}
-                  , etc).
-                </p>
-                <Grid>
-                  <Grid.Column xs={12} md={6}>
-                    {!generatedCodes && (
+          {!generatedCodes && (
+            <Form onSubmit={this.onSubmit} id="generate-2fa-backup-codes">
```
Comment (shrirupa) [inline src/pages/profile/components/BackupCodesModal.js:51]: rearranged the JSX a bit to separate out Form from the regular Panel

**Comment by nstanard** [comment]
Insert Snippet Modal also doesn't properly submit to the tracking event handler.

image

I vote we leave this PR open until I'm able to verify all the Form instances are submitting to the tracking event handler.

**Comment by shrirupa** [comment]
> Insert Snippet Modal also doesn't properly submit to the tracking event handler.
> 
> image
> 
> I vote we leave this PR open until I'm able to verify all the Form instances are submitting to the tracking event handler.

I think we can cut a separate ticket, if we find more cases. This ticket can be limited to addressing the three forms mentioned in the AC. I can handle the InsertSnippetModal in this ticket(it is peculiar since both form and button have the same function attached to onSubmit and onClick. So removing onClick would make it work.)

File: src/pages/profile/components/BackupCodesModal.js [hunk 2]
```diff
@@ -94,14 +76,48 @@ export default class BackupCodesModal extends Component {
...
+              <Panel.LEGACY.Section>
+                <ButtonWrapper>
+                  <Button variant="primary" type="submit" disabled={pending}>
+                    {pending ? 'Generating...' : 'Generate'}
+                  </Button>
+                  <Button variant="secondary" onClick={onClose}>
+                    Cancel
+                  </Button>
+                </ButtonWrapper>
+              </Panel.LEGACY.Section>
```
Comment (nicklemmon) [inline src/pages/profile/components/BackupCodesModal.js:92]: Nice. That `renderButtons()` method seemed unnecessary.

**Review by nicklemmon — APPROVED** [review]
@shrirupa Looks good!

**Pull request merged**